docs/published/REPLACEMASTERS.md

```diff
@@ -0,0 +1,365 @@
+
+- We will be automatically creating appropriate IAM profiles for the Masters to provide access to S3 instead of using Secret and Access keys.
+
+# Prerequisites:
+Terraform, AWS cloud credentials, SSH keys
+
+## Installing Terraform.
+If you're on a Mac environment with [homebrew](https://brew.sh/) installed, simply run the following command:
+```bash
+brew install terraform
+```
+
+Once this command completes, you should be able to run the following command and see output consistent with the version of Terraform you have installed:
+```bash
+$ terraform version
+Terraform v0.11.8
+```
+
+For help installing Terraform on a different OS, please see [here](https://www.terraform.io/downloads.html):
+
+## Ensure you have your AWS Cloud Credentials Properly Set up
+Please follow the AWS guide [Configuring the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html) how to setup your credentials.
+
+## Add your SSH keys to your ssh agent:
+
+Terraform requires SSH access to the instances you launch as part of your DC/OS cluster. As such, we need to make sure that the SSH key used to SSH to these instances is added to your `ssh-agent`, prior to running `terraform`.
+
+If you need help on creating an SSH key-pair for AWS prior to running the command below, please follow the instructions [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html).
+
+Otherwise, just run the following command to add your key to the `ssh-agent`:
+
+```bash
+ssh-add <path_to_your_private_aws_ssh_key>
+```
+
+For Example:
+```bash
+ssh-add ~/.ssh/aws-id-rsa
+```
+
+## Enterprise Edition
+
+DC/OS Enterprise Edition also requires a valid license key provided by Mesosphere that we will pass into our `main.tf` as `dcos_license_key_contents`. For this guide we are going to use the default superuser and password to login:
+
+Username: `bootstrapuser`
+Password: `deleteme`
+
+Please note that this should *NOT* be used in a Production environment and you will need generate a password hash.
+
+# Creating a Cluster
+
+1) Let's stary by creating and local folder and cd'ing into. This folder will be used as the staging ground for downloading all required Terraform modules and holding the configuration for the cluster you are about to create. 
+```
+mkdir dynamic-masters && cd dynamic-masters  
+```
+
+2) Once that is done, copy and paste the example code below into a new file and save it as `main.tf` in the newly created folder.
+
+This example codes tells Terrfaorm to create a DC/OS Enterprise 1.12.0 cluster on AWS with:
+- 3 Masters
+- 2 Private Agents
+- 1 Public Agent
+- With Replaceable Masters enabled
+
+The following lines are required to enable Replaceable Masters:
+
+Example:
+```
+  dcos_exhibitor_explicit_keys   = "false"
+  dcos_exhibitor_storage_backend = "aws_s3"
+  dcos_s3_prefix                 = "exhibitor"
```

Comment:
Another option that we can do here is to make default variables for `dcos_s3_prefix` `dcos_s3_bucket` evaluate `dcos_aws_region` from the region that they have selected via the provider. 

If they do provide a value, it would overwrite the defaults that are set for them when the s3 bucket is being created. This will simplify the main.tf code. It would be good to have conditional s3 bucket creation as well if other modules plan to reuse the infrastructure module.